### PR TITLE
fix: Preventing multiple 'Open in CodeSandbox' buttons being added when switching themes

### LINKED ISSUE #29846: [Bug]: Switching themes in the docsite adds "open in codesandbox" buttons to the examples
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
N/A
```

### Are you reporting Accessibility issue?

no

### Reproduction

react.fluentui.dev

## Actual Behavior
image

## Expected Behavior
to not add buttons

### Logs

_No response_

### Requested priority

Normal

### Products/sites affected

our docsite for v9

### Are you willing to submit a PR to fix?

yes

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

### PR DESCRIPTION
**Pull request by khmakoto**
## Previous Behavior

Switching between themes would make additional `Open in CodeSandbox` buttons to be prepended to each story container.

image

## New Behavior

We now make sure that only one `Open in CodeSandbox` button exists at any time in the container.

image

## Related Issue(s)

- Fixes #29846

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/3f77b502c9e3d91a27ddf6da48ff2ab9b666b91c'>3f77b502c9e3d91a27ddf6da48ff2ab9b666b91c</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

File: packages/react-components/react-storybook-addon-export-to-sandbox/src/sandbox-factory.ts [hunk 1]
```diff
@@ -35,6 +35,14 @@ function addActionButton(container: HTMLElement, config: Data, classList: string
   const files = scaffold[config.bundler](config);
   const action = actionConfig[config.provider];
 
+  // Check if button does not exist already in container
+  for (let i = 0; i < container.children.length; i++) {
+    const child = container.children[i];
+    if (child.tagName === 'BUTTON' && child.textContent === 'Open in CodeSandbox') {
```
Comment (sopranopillow) [inline packages/react-components/react-storybook-addon-export-to-sandbox/src/sandbox-factory.ts:41]: Will this work with the StackBlitz option? it does seem like we test for it: https://github.com/microsoft/fluentui/blob/3f77b502c9e3d91a27ddf6da48ff2ab9b666b91c/packages/react-components/react-storybook-addon-export-to-sandbox/src/sandbox-factory.spec.ts#L98

Comment (khmakoto) [reply packages/react-components/react-storybook-addon-export-to-sandbox/src/sandbox-factory.ts:41]: Oops yeah, checking for `action.label` instead now. Great catch!

**Review by spmonahan — APPROVED** [review]

**Review by TristanWatanabe — COMMENTED** [review]
This seems to be duplicating a check that's already being applied by the `prepareSandboxContainer` function which removes these buttons in dev mode. I'm not aware of what the root cause of the issue is but maybe simply removing the check for dev mode and applying the button removal by default in the code below is the way to go here?

https://github.com/microsoft/fluentui/blob/47f384657cc83d62f766f42057677ad2043de084/packages/react-components/react-storybook-addon-export-to-sandbox/src/sandbox-utils.ts#L34-L37

**Comment by khmakoto** [comment]
> This seems to be duplicating a check that's already being applied by the `prepareSandboxContainer` function which removes these buttons in dev mode. I'm not aware of what the root cause of the issue is but maybe simply removing the check for dev mode and applying the button removal by default in the code below is the way to go here?
> 
> https://github.com/microsoft/fluentui/blob/47f384657cc83d62f766f42057677ad2043de084/packages/react-components/react-storybook-addon-export-to-sandbox/src/sandbox-utils.ts#L34-L37

You're right, I hadn't noticed that. I've removed the check for production now.

File: packages/react-components/react-storybook-addon-export-to-sandbox/src/sandbox-factory.ts [hunk 2]
```diff
@@ -35,6 +35,14 @@ function addActionButton(container: HTMLElement, config: Data, classList: string
   const files = scaffold[config.bundler](config);
   const action = actionConfig[config.provider];
 
+  // Check if button does not exist already in container
+  for (let i = 0; i < container.children.length; i++) {
+    const child = container.children[i];
+    if (child.tagName === 'BUTTON' && child.textContent === `Open in ${action.label}`) {
```
[the remark this answers is not in the document]
Comment (khmakoto) [reply packages/react-components/react-storybook-addon-export-to-sandbox/src/sandbox-factory.ts]: Went with @TristanWatanabe's approach below.

[the remark this answers is not in the document]
Comment (spmonahan) [reply packages/react-components/react-storybook-addon-export-to-sandbox/src/sandbox-factory.ts]: Yeah, that's definitely better :)

**Review by TristanWatanabe — APPROVED** [review]

**Pull request merged**